addons/pos_mrp/models/pos_order.py

```diff
@@ -8,12 +8,14 @@ class PosOrderLine(models.Model):
 
     def _get_stock_moves_to_consider(self, stock_moves, product):
         self.ensure_one()
-        bom = product.env['mrp.bom']._bom_find(product, company_id=stock_moves.company_id.id, bom_type='phantom')[product]
+        bom = product.env['mrp.bom']._bom_find(product, company_id=stock_moves.company_id.id, bom_type='phantom').get(product)
         if not bom:
             return super()._get_stock_moves_to_consider(stock_moves, product)
         _dummy, components = bom.explode(product, self.qty)
         ml_product_to_consider = (product.bom_ids and [comp[0].product_id.id for comp in components]) or [product.id]
-        return stock_moves.filtered(lambda ml: ml.product_id.id in ml_product_to_consider and ml.bom_line_id)
+        if len(_dummy) > 1:
```

Comment:
We usually name a variable `_dummy` when we don't use it. Since it's not the case anymore, you can rename it (simply `boms` for example) to keep the code readable
```suggestion
        if len(boms) > 1:
```

Reply:
Thanks, I will change it.